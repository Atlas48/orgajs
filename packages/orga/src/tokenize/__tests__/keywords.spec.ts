import tok from './tok'

describe('tokenize keywords', () => {
  it('knows keywords', () => {
    expect(tok('#+KEY: Value')).toMatchInlineSnapshot(`
      Array [
        Object {
          "_text": "#+KEY: Value",
          "key": "KEY",
          "type": "keyword",
          "value": "Value",
        },
      ]
    `)
    expect(tok('#+KEY: Another Value')).toMatchInlineSnapshot(`
      Array [
        Object {
          "_text": "#+KEY: Another Value",
          "key": "KEY",
          "type": "keyword",
          "value": "Another Value",
        },
      ]
    `)
    expect(tok('#+KEY: value : Value')).toMatchInlineSnapshot(`
      Array [
        Object {
          "_text": "#+KEY: value : Value",
          "key": "KEY",
          "type": "keyword",
          "value": "value : Value",
        },
      ]
    `)
  })

  it('knows these are not keywords', () => {
    expect(tok('#+KEY : Value')).toMatchInlineSnapshot(`
      Array [
        Object {
          "_text": "#+KEY : Value",
          "type": "text.plain",
          "value": "#+KEY : Value",
        },
      ]
    `)
    expect(tok('#+KE Y: Value')).toMatchInlineSnapshot(`
      Array [
        Object {
          "_text": "#+KE Y: Value",
          "type": "text.plain",
          "value": "#+KE Y: Value",
        },
      ]
    `)
  })
})
