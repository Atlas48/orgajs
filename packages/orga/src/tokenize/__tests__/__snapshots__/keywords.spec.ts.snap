// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tokenize keywords knows keywords 1`] = `
Array [
  Object {
    "key": "KEY",
    "position": Object {
      "end": Object {
        "column": 12,
        "line": 0,
      },
      "start": Object {
        "column": 0,
        "line": 0,
      },
    },
    "type": "keyword",
    "value": "Value",
  },
]
`;

exports[`tokenize keywords knows keywords 2`] = `
Array [
  Object {
    "key": "KEY",
    "position": Object {
      "end": Object {
        "column": 20,
        "line": 0,
      },
      "start": Object {
        "column": 0,
        "line": 0,
      },
    },
    "type": "keyword",
    "value": "Another Value",
  },
]
`;

exports[`tokenize keywords knows keywords 3`] = `
Array [
  Object {
    "key": "KEY",
    "position": Object {
      "end": Object {
        "column": 20,
        "line": 0,
      },
      "start": Object {
        "column": 0,
        "line": 0,
      },
    },
    "type": "keyword",
    "value": "value : Value",
  },
]
`;

exports[`tokenize keywords knows these are not keywords 1`] = `
Array [
  Object {
    "position": Object {
      "end": Object {
        "column": 13,
        "line": 0,
      },
      "start": Object {
        "column": 0,
        "line": 0,
      },
    },
    "type": "text.plain",
    "value": "#+KEY : Value",
  },
]
`;

exports[`tokenize keywords knows these are not keywords 2`] = `
Array [
  Object {
    "position": Object {
      "end": Object {
        "column": 13,
        "line": 0,
      },
      "start": Object {
        "column": 0,
        "line": 0,
      },
    },
    "type": "text.plain",
    "value": "#+KE Y: Value",
  },
]
`;
